src/ir/codegen/gen_iterate_tokens.go

```diff
@@ -13,16 +13,38 @@ func (g *genIterate) Run() error {
 	ctx := g.ctx
 
 	var buf bytes.Buffer
+
+	fmt.Fprintf(&buf, `func handleToken(t *token.Token, cb func(*token.Token) bool) bool {
+	if t == nil {
+		return true
+	}
+	
+	if !cb(t) {
+		return false
+	}
+
+	needReturn := true
+	for _, ff := range t.FreeFloating {
+		needReturn = needReturn && handleToken(ff, cb)
```

Comment:
I'm not a fan of such loops as:
1. When `needReturn` is false, iteration will continue, but `handleToken` will not be called, maybe we can return without it?
2. It's not obvious what is the intended behavior. I can't distinguish it from a bug.

Reply:
Done.